File: server/src/main/com/broll/gainea/server/core/actions/ReactionActions.java
package com.broll.gainea.server.core.actions;

import com.broll.gainea.server.core.player.Player;

public interface ReactionActions {

    void endTurn();

    void sendBoardUpdate();

    ActionContext requireAction(Player player, RequiredActionContext action);

    void sendGameUpdate(Object update);

}

File: server/src/main/com/broll/gainea/server/core/actions/ReactionActions.kt
package com.broll.gainea.server.core.actions

import com.broll.gainea.net.NT_Action
import com.broll.gainea.server.core.player.Player

interface ReactionActions {
    fun endTurn()
    fun sendBoardUpdate()
    fun requireAction(player: Player, action: RequiredActionContext<out NT_Action>): ActionContext<out NT_Action>
    fun sendGameUpdate(update: Any)
}
